src/en/rizzcomic/src/eu/kanade/tachiyomi/extension/en/rizzcomic/UrlUtils.kt

```diff
@@ -0,0 +1,54 @@
+package eu.kanade.tachiyomi.extension.en.rizzcomic
+
+import okhttp3.HttpUrl.Companion.toHttpUrl
+
+object UrlUtils {
+    private fun randomString(length: Int): String {
+        val charPool = ('a'..'z') + ('0'..'9')
+        return List(length) { charPool.random() }.joinToString("")
+    }
+
+    fun generateSeriesLink(seriesId: Int): String {
+        return buildString {
+            append(randomString(4))
+            append("s")
+            append(randomString(4))
+            append(seriesId.toString().padStart(5, '0'))
+            append(randomString(12))
+        }
+    }
+
+    fun generateChapterLink(seriesId: Int, chapterId: Int): String {
+        return buildString {
+            append(randomString(4))
+            append("c")
+            append(randomString(4))
+            append(seriesId.toString().padStart(5, '0'))
+            append(randomString(4))
+            append(chapterId.toString().padStart(6, '0'))
+            append(randomString(4))
+        }
+    }
+
+    // private val seriesRegex = """^[a-z0-9]{4}s[a-z0-9]{4}(\d{5})[a-z0-9]+$""".toRegex()
+    //
+    // fun extractSeriesId(url: String): Int? {
+    //    val path = url.toHttpUrl().pathSegments.last()
+    //    return seriesRegex.find(path)?.groupValues?.get(1)?.toIntOrNull()
+    // }
+
+    private val ChaptersRegex = """^[a-z0-9]{4}c[a-z0-9]{4}(\d{5})[a-z0-9]{4}(\d{6})[a-z0-9]+$""".toRegex()
```

Comment:
Length of last section is known?
```suggestion
    private val ChaptersRegex = """^[a-z0-9]{4}c[a-z0-9]{4}(\d{5})[a-z0-9]{4}(\d{6})[a-z0-9]{4}$""".toRegex()
```

Reply:
Some series links had 12 but others had more than 12 chars, so decided to not assume length. Tho i haven't noticed this with chapter urls